src/inference/src/ie_core.cpp

```diff
@@ -580,6 +580,10 @@ class CoreImpl : public ie::ICore, public std::enable_shared_from_this<ie::ICore
             // as the result is being checked by the user
             strictly_check_dims = false;
         } else {
+            // check if Auto-Batch plugin registered
+            const auto& devices = GetListOfDevicesInRegistry();
+            if (find(devices.begin(), devices.end(), "BATCH") == devices.end())
```

Comment:
registry contains auto batch plugin if OpenVINO is built with ENABLE_AUTO_BATCH.
But if user later removed dll for auto batch, registry still contains information about it, but plugin cannot be loaded.

Reply:
Good catch!

Reply:
This user is a good guy I believe. Changed the code